src/test/java/no/unit/nva/cristin/projects/FetchCristinProjectsTest.java

```diff
@@ -51,39 +51,29 @@
     private static final String TITLE_ILLEGAL_CHARACTERS = "abc123- ,-?";
     private static final String INVALID_JSON = "This is not valid JSON!";
 
-    private static final String ALLOWED_ORIGIN_ENV = "ALLOWED_ORIGIN";
-    private static final String CRISTIN_API_HOST_ENV = "CRISTIN_API_HOST";
-    private static final String CRISTIN_API_DUMMY_HOST = "example.com";
     private static final String ALLOW_ALL_ORIGIN = "*";
     private static final String API_RESPONSE_NON_ENRICHED_PROJECTS_JSON = "api_response_non_enriched_projects.json";
     private static final String API_RESPONSE_ONE_CRISTIN_PROJECT_TO_NVA_PROJECT_JSON =
         "api_response_one_cristin_project_to_nva_project.json";
+    private static final String CRISTIN_GET_PROJECT_RESPONSE = "cristinGetProjectResponse.json";
     private static final ObjectMapper OBJECT_MAPPER = JsonUtils.objectMapper;
     private CristinApiClient cristinApiClientStub;
-    private Environment environment;
+    private Environment environment = new Environment();
     private Context context;
     private ByteArrayOutputStream output;
     private FetchCristinProjects handler;
 
     @BeforeEach
     void setUp() {
-        environment = mock(Environment.class);
-        when(environment.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn(ALLOW_ALL_ORIGIN);
-        when(environment.readEnv(CRISTIN_API_HOST_ENV)).thenReturn(CRISTIN_API_DUMMY_HOST);
-        cristinApiClientStub = new CristinApiClientStub(environment.readEnv(CRISTIN_API_HOST_ENV));
+        cristinApiClientStub = new CristinApiClientStub();
         context = mock(Context.class);
         output = new ByteArrayOutputStream();
         handler = new FetchCristinProjects(cristinApiClientStub, environment);
     }
 
     @ParameterizedTest
     @ArgumentsSource(TestPairProvider.class)
-    void handlerReturnsExpectedBodyWhenRequestInputIsValid(String queryResponse,
-                                                           String getResponse,
-                                                           String expected) throws IOException {
-        cristinApiClientStub = spy(cristinApiClientStub);
-        when(cristinApiClientStub.fetchQueryResults(any())).thenReturn(getReader(queryResponse));
-        when(cristinApiClientStub.fetchGetResult(any())).thenReturn(getReader(getResponse));
+    void handlerReturnsExpectedBodyWhenRequestInputIsValid(String expected) throws IOException {
         var actual = sendDefaultQuery().getBody();
         assertEquals(OBJECT_MAPPER.readTree(expected).toPrettyString(),
```

Comment:
why are you serializing the JsonNodes? compare the JsonNodes directly because the position of the fields plays no role in their equality. This test is based on the fact that OBJECT_MAPPER will always serialize the fields in the same order which is not guaranteed

Reply:
Can you give me an example? I had problems earlier when doing string to string compare, and at that time `readTree()` seemed to fix the issue.

Reply:
just remove the "toPrettyString"